### DISCUSSION TITLE
A Label control text_changed signal.

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#7073] [GUI]
I noticed that the Label control doesn't have a "text_changed" kind of signal.  The bulk of the time, a Label contains static text, however, there are situations where the text for a Label control changes to indicate something.  Using a LineEdit that is not "editable" in this situation may not be necessary/doesn't fit the context, etc, and so a Label control is used.  The ability to be informed when the text changes could be useful in some situations, eg. when your code didn't directly create the control, and there is just no way of providing for this functionality as Label stands at the moment.

  Is this a viable possibility for enhancing/extending the Label class in this way ?

**Comment by AThousandShips** [comment]
But labels aren't changed outside script control? Labels don't change on user input, this doesn't make much sense to me

**Comment by ghost** [comment]
I'm talking about the situation where someone elses' code in a script that you don't have access to changes a label in an interface that you wish to override in some way.

**Comment by AThousandShips** [comment]
That sounds like a very specific case that I don't think is universal enough, try listening to the `draw` signal it will change whenever the text changes

**Comment by ghost** [comment]
That's a good idea.  I'll try that, but I do think a signal for "text_changed" would be better, however, I have given a very specific case.

**Reply by AThousandShips** [reply]
That kind of signal only makes sense when the changes is not in the control of the engine, but from user input, having it for every content that can change from any source isn't really a goal, but however this might be relevant:
* #4868

**Comment by ghost** [comment]
Thanks for that.  It worked really well, the "draw" signal connection.  I thought I'd tried that option, but anyway...

**Reply by AThousandShips** [reply]
Please use the thread comment functionality when it is availble, this thread is full of single comments that makes it very cluttered and hard to follow
